### PR TITLE
Log warning for empty run in terraform_tgw_attachments

### PR DESCRIPTION
**Pull request by hemslo**
Log warning for empty run in terraform_tgw_attachments, so we can disable the integration to save resource. [APPSRE-7624](https://issues.redhat.com/browse/APPSRE-7624)

### <samp>🤖 Generated by Copilot at efa6673</samp>

This pull request improves the test coverage and performance of the `reconcile/terraform_tgw_attachments.py` integration. It adds more tests and mocks to the `reconcile/test/test_terraform_tgw_attachments.py` file and adds a check to skip the integration if there are no participating AWS accounts.

### <samp>🤖 Generated by Copilot at efa6673</samp>

*  Add early return and warning log when no AWS accounts are participating in the integration ([link](https://github.com/app-sre/qontract-reconcile/pull/3602/files?diff=unified&w=0#diff-b2a48a56ab3feb0903c9b33c5eec954718a01524b0dd49a99d882637bb2cc226R372-R377))
*  Add mock for logging module and verify expected log messages in different scenarios ([link](https://github.com/app-sre/qontract-reconcile/pull/3602/files?diff=unified&w=0#diff-b5ab464582f4a599b9e05939387fe44570d983cb1a58d777a7af990bfb3124d4R564-R566))
*  Refactor test_dry_run function into test_empty_run and test_dry_run, and use parameters to customize mocked data for different test cases ([link](https://github.com/app-sre/qontract-reconcile/pull/3602/files?diff=unified&w=0#diff-b5ab464582f4a599b9e05939387fe44570d983cb1a58d777a7af990bfb3124d4L574-R618), [link](https://github.com/app-sre/qontract-reconcile/pull/3602/files?diff=unified&w=0#diff-b5ab464582f4a599b9e05939387fe44570d983cb1a58d777a7af990bfb3124d4L598-R646))
*  Assert no AWS API, OCM, or Terraform calls are made when cluster only has VPC connection ([link](https://github.com/app-sre/qontract-reconcile/pull/3602/files?diff=unified&w=0#diff-b5ab464582f4a599b9e05939387fe44570d983cb1a58d777a7af990bfb3124d4L737-R787))
*  Use tgw_account fixture to provide mocked AWS account object for various test cases, and pass it to _setup_mocks function call to customize mocks ([link](https://github.com/app-sre/qontract-reconcile/pull/3602/files?diff=unified&w=0#diff-b5ab464582f4a599b9e05939387fe44570d983cb1a58d777a7af990bfb3124d4R966), [link](https://github.com/app-sre/qontract-reconcile/pull/3602/files?diff=unified&w=0#diff-b5ab464582f4a599b9e05939387fe44570d983cb1a58d777a7af990bfb3124d4R975), [link](https://github.com/app-sre/qontract-reconcile/pull/3602/files?diff=unified&w=0#diff-b5ab464582f4a599b9e05939387fe44570d983cb1a58d777a7af990bfb3124d4R991), [link](https://github.com/app-sre/qontract-reconcile/pull/3602/files?diff=unified&w=0#diff-b5ab464582f4a599b9e05939387fe44570d983cb1a58d777a7af990bfb3124d4R1000), [link](https://github.com/app-sre/qontract-reconcile/pull/3602/files?diff=unified&w=0#diff-b5ab464582f4a599b9e05939387fe44570d983cb1a58d777a7af990bfb3124d4R1016), [link](https://github.com/app-sre/qontract-reconcile/pull/3602/files?diff=unified&w=0#diff-b5ab464582f4a599b9e05939387fe44570d983cb1a58d777a7af990bfb3124d4R1026), [link](https://github.com/app-sre/qontract-reconcile/pull/3602/files?diff=unified&w=0#diff-b5ab464582f4a599b9e05939387fe44570d983cb1a58d777a7af990bfb3124d4R1043), [link](https://github.com/app-sre/qontract-reconcile/pull/3602/files?diff=unified&w=0#diff-b5ab464582f4a599b9e05939387fe44570d983cb1a58d777a7af990bfb3124d4R1053), [link](https://github.com/app-sre/qontract-reconcile/pull/3602/files?diff=unified&w=0#diff-b5ab464582f4a599b9e05939387fe44570d983cb1a58d777a7af990bfb3124d4R1071), [link](https://github.com/app-sre/qontract-reconcile/pull/3602/files?diff=unified&w=0#diff-b5ab464582f4a599b9e05939387fe44570d983cb1a58d777a7af990bfb3124d4R1080))

### CODE REVIEW

**Review by patjlm — APPROVED** [review]

File: reconcile/terraform_tgw_attachments.py [hunk 1]
```diff
@@ -369,6 +369,12 @@ def run(
     desired_state_data_source = _fetch_desired_state_data_source(account_name)
     accounts = [a.dict(by_alias=True) for a in desired_state_data_source.accounts]
 
+    if not accounts:
+        logging.warning(
+            f"No participating aws accounts found, consider disable this integration, account name: {account_name}"
```
Comment (kwilczynski) [inline reconcile/terraform_tgw_attachments.py:374]:
```suggestion
            f"No participating AWS accounts found, consider disabling this integration, account name: {account_name}"
```

Or

```suggestion
            f"No participating AWS accounts found for {account_name}, consider disabling this integration."
```

Comment (hemslo) [reply reconcile/terraform_tgw_attachments.py:374]: Thanks @kwilczynski ! I'll update tests too

Comment (hemslo) [reply reconcile/terraform_tgw_attachments.py]: updated

**Pull request merged**